ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ExecutionContext.java

```diff
@@ -0,0 +1,24 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.ml.engine.algorithms.remote;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class ExecutionContext {
```

Comment:
Let's add some comment about what is this class for?

Reply:
sure